docs/RELEASES.md

```diff
@@ -26,3 +26,25 @@ This process is slightly different from the above, since we don't necessarily wa
 3. Select `main` or your dev branch from the "Use workflow from" dropdown.
 4. Select your options and click "Run workflow".
 5. Monitor the workflow to ensure the process has succeeded.
+
+## Understanding and Debugging Builder and SDM Releases
+
+### How Connector Builder uses SDM/CDK
+
+The Connector Builder (written in Java) calls the CDK Python package directly, executing the CDK's Source Declarative Manfiest code via Python processes on the Builder container. (The Connector Builder does not directly invoke the SDM image, but there is an open project to change this in the future.)
+
+Our publish flow sends a PR to the Builder repo (`airbyte-platform-internal`) to bump the version used in Builder. The Marketplace Contributions team (aka Connector Builder maintainers) will review and merge the PR.
+
+### How the SDM Image is used in Platform
+
+The platform scans DockerHub at an [every 10 minutes cadence](https://github.com/airbytehq/airbyte-platform-internal/blob/d744174c0f3ca8fa70f3e05cca6728f067219752/oss/airbyte-cron/src/main/java/io/airbyte/cron/jobs/DeclarativeSourcesUpdater.java) as of 2024-12-09. Based on that DockerHub scan, the platform bumps the default SDM version that is stored in the `declarative_manifest_image_version` table in prod.
+
+Note: Currently we don't pre-test images in Platform so manual testing is needed.
+
+### How to confirm what SDM version is used on the Platform
+
+TODO: ...
```

Comment:
```suggestion
Currently there are two ways to do this.

The first option is to look in the `declarative_manifest_image_version` database table  in Prod.

If that is not available as an option, you can run an Builder-created connector in Cloud and note the version number printed in the logs. Warning: this may not be indicative if that connector instance has been manually pinned to a specific version.

TODO: Would be great to find a way to inspect directly without requiring direct prod DB access. 
```